### PR TITLE
Bidirectional search using BFS in an undirected graph in c++ #3187

### PR DESCRIPTION
**Pull request by ManuGupta9780**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Program to perform Bidirectional search using BFS in an undirected graph in c++.

### Checklist

- [ yes] I've read the contribution guidelines.
- [ yes] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Related issue: #3187

### CODE REVIEW

**Comment by ManuGupta9780** [comment]
Kindly review this and let me know the feedback.
@HarshCasper @iamrajiv
@bhav09 @siddharth25pandey

File: C-Plus-Plus/graphs/Bidirectional_search_using_BFS_in_undirected_graph.cpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+Path exist between 3 and 10
+Intersection at: 7
+3 5 6 7 8 10
+
+Time Complexity : O(bd/2)
+Space Complexity : O(bd/2)
+
+{ b = branching factor of tree
+  d = distance of goal vertex from source }
+*/
```
Comment (bhav09) [inline C-Plus-Plus/graphs/Bidirectional_search_using_BFS_in_undirected_graph.cpp]: Add a new line here.

File: C-Plus-Plus/graphs/Bidirectional_search_using_BFS_in_undirected_graph.cpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        path.push_back(parentB[i]);
+        i = parentB[i];
+    }
+
+    for (auto x : path)
+        cout << x << " ";
+    cout << endl;
+}
+
+int bidirectional_search(int sv, int tv, vector<int> graph[], int n, vector<bool>& visitedF,
```
Comment (bhav09) [inline C-Plus-Plus/graphs/Bidirectional_search_using_BFS_in_undirected_graph.cpp]: Lacks comment. 
Comments should be above the code segment and not alongside.

File: C-Plus-Plus/graphs/Bidirectional_search_using_BFS_in_undirected_graph.cpp [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        if (flag == 1)
+            inode = i;
+
+        if (inode != -1)
+            return inode;
+    }
+    return -1;
+}
+
+int main()
```
Comment (bhav09) [inline C-Plus-Plus/graphs/Bidirectional_search_using_BFS_in_undirected_graph.cpp:154]: The main function should act as the driver function, cut short the length of this function by transposing the redundant to a feature/helper function.

**Review by bhav09 — CHANGES_REQUESTED** [review]
Reviewed.

**Comment by ManuGupta9780** [comment]
@bhav09  I have made the requested changes. Kindly review it.

**Comment by bhav09** [comment]
Great work! @ManuGupta9780

**Comment by ManuGupta9780** [comment]
@bhav09  Thanks:)

**Pull request merged**